### PR TITLE
Add IsFolderWorkerSpace wrapper to ISolutionInfoProvider

### PR DESCRIPTION
**Pull request by bigfluffycookie**
Remove calls to GetService from Importing Constructor.

part of https://github.com/SonarSource/sonarlint-visualstudio/issues/4842

### CODE REVIEW

File: src/Infrastructure.VS.UnitTests/SolutionInfoProviderTests.cs [hunk 1]
```diff
@@ -146,6 +146,36 @@ public void GetSolutionFullyOpen_ReturnsExpectedValue(bool isFullyOpen)
             actual.Should().Be(isFullyOpen);
         }
 
+        [TestMethod]
```
Comment (bigfluffycookie) [inline src/Infrastructure.VS.UnitTests/SolutionInfoProviderTests.cs:149]: These two tests are almost the same as I deleted in FolderWorkspaceServiceTests

File: src/Infrastructure.VS/SolutionInfoProvider.cs [hunk 2]
```diff
@@ -87,6 +89,35 @@ public bool IsSolutionFullyOpened()
...
+        public bool IsFolderWorkspace()
+        {
+            var isFolderWorkspace = false;
+            threadHandling.RunOnUIThread(() => isFolderWorkspace = GetIsFolderWorkspace());
+            return isFolderWorkspace;
+        }
+
+        private bool GetIsFolderWorkspace()
+        {
+            var solution = serviceProvider.GetService(typeof(SVsSolution)) as IVsSolution;
```
Comment (bigfluffycookie) [inline src/Infrastructure.VS/SolutionInfoProvider.cs:108]: Copied over from FolderWorkspace class

File: src/Infrastructure.VS/SolutionInfoProvider.cs [hunk 3]
```diff
@@ -87,6 +89,35 @@ public bool IsSolutionFullyOpened()
...
+        }
+
+        public bool IsFolderWorkspace()
+        {
+            var isFolderWorkspace = false;
+            threadHandling.RunOnUIThread(() => isFolderWorkspace = GetIsFolderWorkspace());
+            return isFolderWorkspace;
+        }
+
+        private bool GetIsFolderWorkspace()
```
Comment (bigfluffycookie) [inline src/Infrastructure.VS/SolutionInfoProvider.cs:106]: Copied over from FolderWorkspace class

**Review by duncanp-sonar — APPROVED** [review]
Nice!

**Pull request merged**